src/App.tsx

```diff
@@ -28,21 +29,34 @@ const PullRequestPage = lazy(() => import('./pages/PullRequestPage'))
 const RepoPage = lazy(() => import('./pages/RepoPage'))
 const SyncProviderPage = lazy(() => import('./pages/SyncProviderPage'))
 
+interface URLParams {
+  provider: string
+}
+
 const HomePageRedirect = () => {
-  const { provider } = useParams()
+  const { provider } = useParams<URLParams>()
   const { data: currentUser } = useUser()
+  const { data: internalUser } = useInternalUser({})
   const { params } = useLocationParams()
 
-  if (!provider || !currentUser) {
-    return <Redirect to="/login" />
+  let redirectURL = '/login'
+
+  if (internalUser) {
+    redirectURL = `/${internalUser.owners![0]?.service}/${
```

Comment:
if internalUser exists, my assumption here is owners will at least be an empty array

Reply:
I can't find any reason why that assumption would be bad. but just in case you didn't know how to optional chaining with array indexing (like I didn't until recently lol):
```suggestion
    redirectURL = `/${internalUser.owners?.[0]?.service}/${
```

I think that would be a safer play. Generally try to steer clear of `!` myself - it's scary 😰